### DISCUSSION TITLE
Error  Webpack in react Module not found: Error: Can't resolve 'https'

### DISCUSSION THREAD

**Discussion by lnvestor** [webpack/webpack#15017] [Q&A]
# Bug report

I am a new developer with web3 js. I work with "react": "^17.0.2", So I just added the library on my dependencies (npm i web3) But when I try to instantiate my web3 object like this web3 = new Web3((window as any).ethereum); Or like this web3 = new Web3();

And I try to start my app (npm start)

I get a lot of errors

about my webpack I have to add some fallback in my webpack.config.js

**What is the current behavior?**

**ERROR in ./node_modules/cipher-base/index.js 3:16-43

Module not found: Error: Can't resolve 'stream' in 'C:...\node_modules\cipher-base'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default. This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:

add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
install 'stream-browserify' If you don't want to include a polyfill, you can use an empty module like this: resolve.fallback: { "stream": false }
ERROR in ./node_modules/eth-lib/lib/bytes.js 9:193-227

Module not found: Error: Can't resolve 'crypto' in ....

ERROR in ./node_modules/ethereumjs-util/dist.browser/account.js 71:31-48

Module not found: Error: Can't resolve 'assert' in ...

ERROR in ./node_modules/ethereumjs-util/dist.browser/address.js 14:31-48

Module not found: Error: Can't resolve 'assert' in ... ERROR in ./node_modules/ethereumjs-util/dist.browser/object.js 46:31-48

Module not found: Error: Can't resolve 'assert' in ...

ERROR in ./node_modules/web3-eth-accounts/lib/index.js 31:74-91

Module not found: Error: Can't resolve 'crypto' in ... ERROR in ./node_modules/web3-eth-accounts/node_modules/eth-lib/lib/bytes.js 7:193-227

Module not found: Error: Can't resolve 'crypto' in ...

ERROR in ./node_modules/web3-providers-http/lib/index.js 30:11-26

Module not found: Error: Can't resolve 'http' in ...

ERROR in ./node_modules/web3-providers-http/lib/index.js 32:12-28

Module not found: Error: Can't resolve 'https' in ...

ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 37:11-26

Module not found: Error: Can't resolve 'http' in ...

ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 39:12-28

Module not found: Error: Can't resolve 'https' in ...

ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 41:9-22

Module not found: Error: Can't resolve 'os' in 'C:...\node_modules\xhr2-cookies\dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default. This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:

add a fallback 'resolve.fallback: { "os": require.resolve("os-browserify/browser") }'
install 'os-browserify' If you don't want to include a polyfill, you can use an empty module like this: resolve.fallback: { "os": false }**

**If the current behavior is a bug, please provide the steps to reproduce.**

npx create-react-app my-app
cd my-app
install moralis provider
error

**What is the expected behavior?**

**Other relevant information:**
webpack version:  5
Node.js version: 16
Operating System:  win11
Additional tools: Morlais

**Comment by alexander-akait** [comment]
For `http` and `https` protocols you need to setup https://webpack.js.org/configuration/experiments/#experimentsbuildhttp, it is experiment

**Reply by alexander-akait** [reply]
If you need to polyfill Node.js packages you need to use `resolve.alias`, we don't polyfill Node.js packages by default, here is answer https://stackoverflow.com/questions/64557638/how-to-polyfill-node-core-modules-in-webpack-5
